### PR TITLE
Corrected a mistake in the documentation for the Spring Architecture

### PR DESCRIPTION
**Pull request by angel-gruevski**
Corrected a mistake in the documentation for the Spring Architecture where it was mentioned that RequestCacheAwareFilter is used to save the HttpServletRequest instead of the ExceptionTranslationFilter

The issue id for this pull request is 14855

https://github.com/spring-projects/spring-security/issues/14855

### CODE REVIEW

File: docs/modules/ROOT/pages/servlet/architecture.adoc [hunk 1]
```diff
@@ -449,7 +449,7 @@ In Spring Security this is done by saving the `HttpServletRequest` using a <<req
 
 The `HttpServletRequest` is saved in the {security-api-url}org/springframework/security/web/savedrequest/RequestCache.html[`RequestCache`].
 When the user successfully authenticates, the `RequestCache` is used to replay the original request.
-The <<requestcacheawarefilter,`RequestCacheAwareFilter`>> is what uses the `RequestCache` to save the `HttpServletRequest`.
+The `RequestCacheAwareFilter` uses the `RequestCache` to get the saved `HttpServletRequest` after the user authenticates, while the `ExceptionTranslationFilter` uses the `RequestCache` to save the `HttpServletRequest` after it detects `AuthenticationException`, before redirecting the user to the login endpoint.
```
Comment (jzheaux) [inline docs/modules/ROOT/pages/servlet/architecture.adoc:452]: Is there a reason to remove the link?

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @angel-gruevski! In addition to my inline feedback, will you please adjust your Git commit message to follow the following format:

```
Correct Documentation

Your description, if any, wrapping about every 60 characters
or so.

Closes gh-14855
```

That syntax helps so that, first, the associated ticket is closed automatically, and second, the Git history is easier to read.

File: docs/modules/ROOT/pages/servlet/architecture.adoc (hunk 1 above)
Comment (angel-gruevski) [reply docs/modules/ROOT/pages/servlet/architecture.adoc]: Not really, it seems I've just deleted the whole link section and replaced with the filter name. I've corrected this, thanks for noticing.

**Comment by jzheaux** [comment]
@angel-gruevski, thanks so much for the updates. Will you please do one more cleanup item for me and squash your commits?

**Comment by marcusdacoregio** [comment]
Thanks @angel-gruevski. This is now merged into `5.8.x` via https://github.com/spring-projects/spring-security/commit/ce995a01789e70516ec128ae7092bb8438b8bb89 and forward-ported to `6.1.x`, `6.2.x` and `main`

**Pull request closed without merging**